allure-kotlin-android/src/main/java/io/qameta/allure/android/AllureScreenshot.kt

```diff
@@ -0,0 +1,42 @@
+package io.qameta.allure.android
+
+import android.app.UiAutomation
+import android.graphics.Bitmap
+import android.util.Log
+import androidx.annotation.IntRange
+import androidx.test.platform.app.InstrumentationRegistry
+import io.qameta.allure.kotlin.Allure
+import java.io.ByteArrayOutputStream
+import java.io.InputStream
+
+private const val TAG = "AllureScreenshot"
+
+/**
+ * Takes a screenshot of a device in a given [quality] and attaches it to current step or test run.
+ * Quality must be in range [0..100] or an exception will be thrown
+ *
+ * It uses [android.app.UiAutomation] under the hood to take the screenshot (just like the UiAutomator).
+ *
+ * @return true if screen shot is created and attached successfully, false otherwise
+ */
+@Suppress("unused")
+fun allureScreenshot(name: String = "screenshot", @IntRange(from = 0, to = 100) quality: Int = 90): Boolean {
+    require(quality in (0..100)) { "quality must be 0..100" }
+    val uiAutomation = InstrumentationRegistry.getInstrumentation().uiAutomation ?: return false.also {
```

Comment:
I have just noticed that `uiAutomation` requires API 18. I am migrating to `uiAutomator` to have the support since the lower versions of the API.